include/alpaka/warp/Traits.hpp

```diff
@@ -0,0 +1,194 @@
+/* Copyright 2020 Sergei Bastrakov
+ *
+ * This file is part of Alpaka.
+ *
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/.
+ */
+
+#pragma once
+
+#include <alpaka/core/Common.hpp>
+#include <alpaka/core/Concepts.hpp>
+
+#include <cstdint>
+#include <type_traits>
+
+namespace alpaka
+{
+    //-----------------------------------------------------------------------------
+    //! The thread warp specifics
+    namespace warp
+    {
+        struct ConceptWarp{};
+
+        //-----------------------------------------------------------------------------
+        //! The warp traits.
+        namespace traits
+        {
+            //#############################################################################
+            //! The warp size trait.
+            template<
+                typename TWarp,
+                typename TSfinae = void>
+            struct GetSize;
+
+            //#############################################################################
+            //! The all warp vote trait.
+            template<
+                typename TWarp,
+                typename TSfinae = void>
+            struct All;
+
+            //#############################################################################
+            //! The any warp vote trait.
+            template<
+                typename TWarp,
+                typename TSfinae = void>
+            struct Any;
+
+            //#############################################################################
+            //! The ballot warp vote trait.
+            template<
+                typename TWarp,
+                typename TSfinae = void>
+            struct Ballot;
+
+            //#############################################################################
+            //! The active mask trait.
+            template<
+                typename TWarp,
+                typename TSfinae = void>
+            struct Activemask;
+        }
+
+        //-----------------------------------------------------------------------------
+        //! Returns warp size.
+        //!
+        //! \tparam TWarp The warp implementation type.
+        //! \param warp The warp implementation.
+        ALPAKA_NO_HOST_ACC_WARNING
+        template<
+            typename TWarp>
+        ALPAKA_FN_ACC auto getSize(
+            TWarp const & warp)
+        -> std::int32_t
```

Comment:
Suggestion: we should remove the return value here, this will give the compiler the chance to use 8Bit or 16Bit depending of what is returned in the implementation.

Reply:
I don't really like the suggestion (again, a user has to do type casts, as it's not even clear whether the result is signed or unsigned). But I guess this is logical given a similar change was already done for `activemask()` and `ballot()`. Will do.